x-pack/test/functional/apps/lens/chart_data.ts

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { DebugState } from '@elastic/charts';
+import expect from '@kbn/expect';
+import { range } from 'lodash';
+import { FtrProviderContext } from '../../ftr_provider_context';
+
+export default function ({ getService, getPageObjects }: FtrProviderContext) {
+  const PageObjects = getPageObjects(['visualize', 'lens', 'common', 'header']);
+  const elasticChart = getService('elasticChart');
+
+  describe('lens chart data', () => {
+    before(async () => {
+      await PageObjects.visualize.navigateToNewVisualization();
+      await PageObjects.visualize.clickVisType('lens');
+      await elasticChart.setNewChartUiDebugFlag(true);
+      await PageObjects.lens.goToTimeRange();
+
+      await PageObjects.lens.configureDimension({
+        dimension: 'lnsXY_xDimensionPanel > lns-empty-dimension',
+        operation: 'terms',
+        field: 'ip',
+      });
+
+      await PageObjects.lens.configureDimension({
+        dimension: 'lnsXY_yDimensionPanel > lns-empty-dimension',
+        operation: 'avg',
+        field: 'bytes',
+      });
+
+      await PageObjects.header.waitUntilLoadingHasFinished();
+    });
+
+    const expectedData = [
+      { x: '0.53.251.53', y: 4624.75 },
+      { x: '0.108.3.2', y: 7359.41 },
+      { x: '0.209.80.244', y: 6169.9 },
+      { x: '0.228.1.71', y: 7092.8 },
+      { x: '0.254.91.215', y: 3835.58 },
+      { x: '__other__', y: 5727.24 },
+    ];
+
+    function assertMatchesExpectedData(state: DebugState) {
+      expect(
+        state.bars![0].bars.map((bar) => ({
+          x: bar.x,
+          y: Math.round(bar.y * 100) / 100,
```

Comment:
Could you comment on why this rounding is needed? Or maybe the `expect` library has a number precision parameter?

Reply:
it's because the data table rounds numbers to two decimal points, but the xy chart does not